File: sqldelight-sample/src/main/java/com/example/sqldelight/hockey/ui/PlayerRow.java
package com.example.sqldelight.hockey.ui;

import android.content.Context;
import android.util.AttributeSet;
import android.widget.LinearLayout;
import android.widget.TextView;
import butterknife.Bind;
import butterknife.ButterKnife;
import com.example.sqldelight.hockey.R;
import com.example.sqldelight.hockey.data.Player;
import com.example.sqldelight.hockey.data.Team;

public final class PlayerRow extends LinearLayout {
  @Bind(R.id.player_name) TextView playerName;
  @Bind(R.id.team_name) TextView teamName;
  @Bind(R.id.player_number) TextView playerNumber;

  public PlayerRow(Context context, AttributeSet attrs) {
    super(context, attrs);
  }

  @Override protected void onFinishInflate() {
    super.onFinishInflate();
    ButterKnife.bind(this);
  }

  public void populate(Player player, Team team) {
    playerName.setText(player.firstName() + " " + player.lastName());
    playerNumber.setText(String.valueOf(player.number()));
    teamName.setText(team.name());
  }
}

File: sample/android/src/main/java/com/example/sqldelight/hockey/ui/PlayerRow.kt
package com.example.sqldelight.hockey.ui

import android.content.Context
import android.util.AttributeSet
import android.widget.LinearLayout
import android.widget.TextView
import com.example.sqldelight.hockey.R
import com.example.sqldelight.hockey.data.ForTeam

class PlayerRow(
  context: Context,
  attrs: AttributeSet
) : LinearLayout(context, attrs) {
  fun populate(row: ForTeam) {
    findViewById<TextView>(R.id.player_name).text = "${row.first_name} ${row.last_name}"
    findViewById<TextView>(R.id.player_number).text = row.number
    findViewById<TextView>(R.id.team_name).text = row.teamName
  }
}
